### PR TITLE
Call out the incompatibility of wildcards and L7 permissions

### PR DESCRIPTION
**Pull request by banks**
There is a limitation on L7 intentions that I couldn't find documented anywhere. This fixes that at least in the reference docs for the affected fields.

### CODE REVIEW

File: website/content/docs/connect/config-entries/service-intentions.mdx
Comment (preetapan) [inline website/content/docs/connect/config-entries/service-intentions.mdx:284]: DO you think "Wild card intentions cannot be used when defining L7 permissions" is less ambiguos wording than "is incompatible" ?

File: website/content/docs/connect/config-entries/service-intentions.mdx
Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/service-intentions.mdx:400]:
```suggestion
                      Setting \`Permissions\` is not valid if a wildcard is used for the \`Name\` or \`Namespace\` because they can only be
```
This is a little grammar nitpick, but a common diction error is using "since" when you mean "because". The "since" preposition relates to time, whereas "because" relates to function. Have some salt with this comment.

File: website/content/docs/connect/config-entries/service-intentions.mdx
Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/service-intentions.mdx:411]:
```suggestion
                      because they can only be applied to services with a compatible protocol.`,
```
See previous comment

**Review by trujillo-adam — APPROVED** [review]
Very minor suggestions

File: website/content/docs/connect/config-entries/service-intentions.mdx
Comment (banks) [reply website/content/docs/connect/config-entries/service-intentions.mdx:284]:
```suggestion
        "The name of the destination service for all intentions defined in this config entry. This may be set to the wildcard character (`*`) to match all services that don't otherwise have intentions defined. Wildcard intentions cannot be used when defining L7 [`Permissions`](/docs/connect/config-entries/service-intentions#permissions).",
```

File: website/content/docs/connect/config-entries/service-intentions.mdx
Comment (banks) [reply website/content/docs/connect/config-entries/service-intentions.mdx:293]:
```suggestion
        "Specifies the namespaces the config entry will apply to. This may be set to the wildcard character (`*`) to match all services in all namespaces that don't otherwise have intentions defined. Wildcard intentions cannot be used when defining L7 [`Permissions`](/docs/connect/config-entries/service-intentions#permissions).",
```

File: website/content/docs/connect/config-entries/service-intentions.mdx
Comment (banks) [reply website/content/docs/connect/config-entries/service-intentions.mdx:327]:
```suggestion
            "The name of the destination service for all intentions defined in this config entry. This may be set to the wildcard character (`*`) to match all services that don't otherwise have intentions defined. Wildcard intentions cannot be used when defining L7 [`Permissions`](/docs/connect/config-entries/service-intentions#permissions).",
```

File: website/content/docs/connect/config-entries/service-intentions.mdx
Comment (banks) [reply website/content/docs/connect/config-entries/service-intentions.mdx:335]:
```suggestion
            "Specifies the namespaces the config entry will apply to. This may be set to the wildcard character (`*`) to match all services in all namespaces that don't otherwise have intentions defined. If not set, the namespace used will depend on the `connectInject.consulNamespaces` configuration. See [ServiceIntentions Special Case (Enterprise)](/docs/k8s/crds#serviceintentions-special-case-enterprise) for more details. Wildcard intentions cannot be used when defining L7 [`Permissions`](/docs/connect/config-entries/service-intentions#permissions).",
```

File: website/content/docs/connect/config-entries/service-intentions.mdx
Comment (banks) [reply website/content/docs/connect/config-entries/service-intentions.mdx:400]: Nice tip thanks!

**Comment by banks** [comment]
Thanks @preetapan that is way more succinct. Thanks Blake for catching the link and Adam for the grammar tip!

**Review by blake — APPROVED** [review]

File: website/content/docs/connect/config-entries/service-intentions.mdx
Comment (mkeeler) [inline website/content/docs/connect/config-entries/service-intentions.mdx:284]: Would it be worthwhile to codify this in validation logic for the config entry write. That way even if you didn't read the docs there would be an error returned by the API indicating the same thing.

Comment (mkeeler) [reply website/content/docs/connect/config-entries/service-intentions.mdx:284]: This seems like a breaking change though so it would probably need to wait for the next major release not to mention having some supporting code for stripping out the Permissions for wildcard entries when they are reinserted into memdb.

Comment (rboyer) [reply website/content/docs/connect/config-entries/service-intentions.mdx:284]: It already behaved this way: https://github.com/hashicorp/consul-enterprise/blob/main/agent/structs/config_entry_intentions.go#L645

Comment (rboyer) [reply website/content/docs/connect/config-entries/service-intentions.mdx:284]: Paul is just calling out how it already worked better on the docs.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/458328.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 020e2692dad905947082453d9bc95a15fd01ebfa onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 020e2692dad905947082453d9bc95a15fd01ebfa onto `release/1.10.x` succeeded!

**Pull request merged**
